### PR TITLE
feat(Text): New Text component; upgrade to styled-system@v5

### PR DESCRIPTION
**Pull request by kylealwyn**
### New `<Text />` component
- `<Text as="h1" fontSize={100} color="primaryLight" />`
- Should allow us to remove many of our one-off styled components, which are typically used only to style some text

### Upgrade styled-system
- Better performance for foundational components like Box and Flex with `compose`
- Better support for responsive styles, e.g. `<Box m={{ _: 0, sm: 12 }} />`, which says for all screens, apply 0 margin, and for small screens and up, apply a 12px margin

### Clean up theme.js
- Better organization
- Remove grid nesting
- Add 24px to space scale
- Allow breakpoints to be overridden

### Clean up `<Column />`
- Remove `order` style so styled-system can apply `order` responsively, e.g. `<Column order={{ _: -1, sm: 1 }} />`
- Prettier wasn't liking some of the implicit ternary returns

### Fix Dropdown stories
- Storybook was not liking hooks in the base story
- Modal was using old Dropdown syntax

### CODE REVIEW

File: rollup.config.js [hunk 1]
```diff
@@ -15,11 +16,18 @@ export default {
       format: 'es',
     },
   ],
-  external: [...Object.keys(pkg.dependencies || {}), ...Object.keys(pkg.peerDependencies || {})],
+  external: [
+    ...Object.keys(pkg.dependencies || {}).filter(dep => dep !== 'styled-components'),
```
Comment (cehsu) [inline rollup.config.js]: Would it be worthwhile to add a comment about why we're doing this?

File: src/Alert/Alert.js [hunk 2]
```diff
@@ -29,10 +30,11 @@ const StyledAlert = createComponent({
 });
 
 /** Alerts are typically used to display meaningful copy to users - typically notifying the user of an important message. */
-const Alert = props => <StyledAlert {...props} />;
+const Alert = React.forwardRef((props, ref) => <StyledAlert {...props} ref={ref} />);
```
Comment (cehsu) [inline src/Alert/Alert.js]: 🙏

File: src/Badge/Badge.js [hunk 3]
```diff
@@ -22,11 +23,12 @@ const StyledBadge = createComponent({
   },
 });
 
-const Badge = props => <StyledBadge {...props} />;
+const Badge = React.forwardRef((props, ref) => <StyledBadge {...props} ref={ref} />);
```
Comment (cehsu) [inline src/Badge/Badge.js]: 🙏

File: src/Alert/Alert.js [hunk 4]
```diff
@@ -29,10 +30,11 @@ const StyledAlert = createComponent({
...
 
 Alert.propTypes = {
   variant: PropTypes.string,
+  ...propTypes.space,
```
Comment (cehsu) [inline src/Alert/Alert.js]: 😎

File: src/Text/Text.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import { createComponent } from '../utils';
+
+const StyledText = createComponent({
+  name: 'Text',
+  tag: 'span',
+  style: compose(
+    space,
+    color,
+    typography
+  ),
```
Comment (cehsu) [inline src/Text/Text.js]: So slick 😎

File: src/Text/Text.spec.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  });
+
+  test('"as" prop', () => {
+    const { asFragment } = renderWithTheme(<Text as="h1">Hey!</Text>);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('"color" prop', () => {
+    const { asFragment } = renderWithTheme(<Text color="red">Hey!</Text>);
+    expect(asFragment()).toMatchSnapshot();
```
Comment (cehsu) [inline src/Text/Text.spec.js]: While we're at it, should we add a `space` prop test as well?

File: src/Text/Text.stories.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+export default {
+  title: 'Components|Text',
+  component: Text,
+};
+
+export const Basic = () => (
+  <>
+    <Text>The basic text component renders as a {'<span />'}</Text>
+    <Text as="h1">It can render as other tags, like this {'<H1/>'}</Text>
+    <Text color="red">It can use colors from the theme</Text>
```
Comment (cehsu) [inline src/Text/Text.stories.js]: Similar to above, should we add a spacing prop case while we're going to the trouble to illustrate these props, or is it similar enough to the existing spacing props on other components, that there's no need for the additional explanation?

File: src/Modal/Modal.dropdownExample.js [hunk 8]
```diff
@@ -27,16 +27,10 @@ export default class ModalDropdownDemo extends React.Component {
     return (
       <div>
         <Dropdown placement="top" width={250} trigger={<Button variant="success">Open Dropdown</Button>}>
-          <Dropdown.Header title="Dropdown" />
```
Comment (cehsu) [inline src/Modal/Modal.dropdownExample.js]: No more need for the multi section dropdown? Do we want to use the divider here? I think we had added the dropdown to facilitate keyboard accessibility testing.

File: src/Grid/Column.js [hunk 9]
```diff
@@ -32,24 +34,26 @@ const getOffset = (offset, columns) => css`
...
+          ${props[`${breakpoint}Offset`] && getOffset(props[`${breakpoint}Offset`], theme.gridColumns)};
+        `;
+      }
+
+      return css`
+        @media (min-width: ${theme.breakpoints[breakpoint]}) {
+          ${getColumnWidth(props[breakpoint], theme.gridColumns)};
+          ${props[`${breakpoint}Offset`] !== undefined && getOffset(props[`${breakpoint}Offset`], theme.gridColumns)};
+        }
+      `;
```
Comment (cehsu) [inline src/Grid/Column.js]: Could we use breakpoint aliases to generate these media queries?

Comment (kylealwyn) [reply src/Grid/Column.js]: I'd rather not touch the implementation in this PR

File: src/Modal/Modal.dropdownExample.js (hunk 8 above)
Comment (kylealwyn) [reply src/Modal/Modal.dropdownExample.js]: wut

File: src/Text/Text.stories.js (hunk 7 above)
Comment (kylealwyn) [reply src/Text/Text.stories.js]: The latter

**Pull request merged**